src/verify.c

```diff
@@ -770,6 +770,29 @@ static enum swupd_code deal_with_extra_files(struct manifest *manifest, bool fix
 	return ret;
 }
 
+static int find_unsafe_to_delete(const void *a, const void *b)
```

Comment:
Can you check if this works for you?

        static int find_unsafe_to_delete(const void *a, const void *b)
        {
                struct file *A, *B;
                int ret;
        
                A = (struct file *)a;
                B = (struct file *)b;
        
                ret = strcmp(A->filename, B->filename);
                if (ret) {
                        return ret < 0 ? -1 : 1;
                }
        
                if (!A->is_deleted) {
                        return -1;
                }
        
                if (B->is_deleted) {
                        return 1;
                }
        
                return 0;
        }

Reply:
🤔 I think I get your idea now... so when both lists are pointing to a matched element, but the condition we want is not met (in this case `A->is_deleted && !B->is_deleted`), we don't necessarily have to move both pointers to the next element, we can just move either one to the next element, and that will force the other pointer to move forward also in the next loop cycle. That way you still only need 3 return codes and not 4, very clever. I couldn't think of a way of making it work without needing 4 RCs. I'll give this a try! 😎